src/Microsoft.AspNet.Mvc.Core/ApplicationModels/DefaultControllerModelBuilder.cs

```diff
@@ -80,6 +85,22 @@ protected virtual ControllerModel CreateControllerModel([NotNull] TypeInfo typeI
             AddRange(controllerModel.Filters, attributes.OfType<IFilter>());
             AddRange(controllerModel.RouteConstraints, attributes.OfType<IRouteConstraintProvider>());
 
+            var enableCors = attributes.OfType<IEnableCorsMetadata>().SingleOrDefault();
```

Comment:
:+1: on the scenario. for restricting at a global level we could add a filter at a global level and that would have restricted, however that is no different how other filters behave. I would add a func test for this nevertheless.